### PR TITLE
Sample's header table fields are not validated on submit

### PR DESCRIPTION
**Pull request by xispa**
## Description of the issue/feature this PR addresses
This Pull Request validates the values submitted for fields displayed in Sample's edit view and displays a message if errors are found.

## Current behavior before PR

The values for fields displayed in the Sample's edit view (header table) are not validated on submit

## Desired behavior after PR is merged

The values for fields displayed in the Sample's edit view (header table) are validated on submit

--
I confirm I have tested this PR thoroughly and coded it according to [PEP8][1]
and [Plone's Python styleguide][2] standards.

[1]: https://www.python.org/dev/peps/pep-0008
[2]: https://docs.plone.org/develop/styleguide/python.html

### CODE REVIEW

File: src/bika/lims/browser/header_table.py [hunk 1]
```diff
@@ -61,28 +61,65 @@ def __call__(self):
...
+                self.request[field.getName()] = value
+
+                # Validate the field values
+                err = field.validate(value, self.context)
+                if err:
+                    # Clear field values and add the error
+                    errors.update({field.getName(): err})
+                    field_values = {}
+
+                elif not errors:
```
Comment (ramonski) [inline src/bika/lims/browser/header_table.py]: shouldn't that be a single `else` ?

**Review by ramonski — CHANGES_REQUESTED** [review]
The browser does only display: "Error Please correct the indicated errors", but no errors are indicated.

**Comment by xispa** [comment]
> The browser does only display: "Error Please correct the indicated errors", but no errors are indicated.

Oops, you were right, missed https://github.com/senaite/senaite.core/pull/1687/commits/3fe84285a92a53877e1de8a901bc76870a5537dd

The display of the error is handled by the widget by getting the value for `error` key as usual. For instance:

```html
<div class="field MyWidget"
          tal:define="values python:field.getEditAccessor(here)();
                      session_value python:here.session_restore_value(fieldName, values);
                      cached_value python:request.get(fieldName, session_values);
                      value python:cached_value or value;
                      required field/required|nothing;
                      error python:errors.get(fieldName, '');
                      error_clazz python: error and 'error ' or ''"
          tal:attributes="class string:${error_clazz} field">

        <div class="fieldErrorBox" tal:content="error"></div>

        <input type="text" size="10" placeholder="My widget text"
               tal:attributes="id string:${fieldName}; 
                                      name string:${fieldName};
                                      value value;"/>
        </div>
      </div>
```

File: src/bika/lims/browser/header_table.py (hunk 1 above)
Comment (xispa) [reply src/bika/lims/browser/header_table.py]: If it was `else`, valid values would be stored on submit, while those not valid would not. I think is better to not store anything unless all fields are valid

Comment (xispa) [reply src/bika/lims/browser/header_table.py]: Umm, you're right, is meaningless beacuse we already check later if there are errors

**Review by ramonski — APPROVED** [review]
Works properly, thanks 👍

**Pull request merged**